initrd/bin/kexec-save-default

```diff
@@ -129,6 +129,55 @@ if [ ! -r $ENTRY_FILE -o ! -r $HASH_FILE ]; then
 	die "Failed to write default config"
 fi
 
+if [ "$save_key" = "y" ]; then
+	# logic to parse OS initrd to extract crypttab, its filepaths and its OS defined options
+	mkdir -p /tmp/initrd_extract
+	cd /tmp/initrd_extract
+	# Get initrd filename selected to be default initrd that OS could be using to configure LUKS on boot by deploying crypttab files
+	current_default_initrd=$(cat /boot/kexec_default_hashes.txt | grep initr | awk -F " " {'print $NF'} | sed 's/\.\//\/boot\//g')
+	
+	# Get crypttab files paths from initrd
+	# We expect zcat to fail on uncompressed file and try alternative cpio-only command if failing, redirecting error output to devnull
+	# To do so we disable exiting if things fail here temporarely
+	echo "Checking current selected default boot's $current_default_initrd for existing crypttab files..."
+	set +e +o pipefail
+	crypttab_files=$(zcat $current_default_initrd 2> /dev/null | cpio --list --quiet | grep crypttab 2> /dev/null) || crypttab_files=$(cpio --list --quiet < $current_default_initrd | grep crypttab 2> /dev/null)
```

Comment:
Saw your note about `set +e +o pipefail`, agree it's a bit complex/fragile :thinking: 

You could break it up into an "optionally decompress" step and then do everything else once, like:

```
# First either decompress or use the original if it's not compressed
initrd_decompressed="/tmp/initrd_decompressed.cpio"
if ! zcat < "$current_default_initrd" > "$initrd_decompressed"; then
    initrd_decompressed="$current_default_initrd" # Use uncompressed image directly
fi

# Then use $initrd_decompressed in the remaining steps - checking for crypttab presence, extracting crypttabs
```

(Used zcat instead of `gzip -d` because the initrd doesn't have a .gz extension.)